src/test/java/org/reborndb/reborn/TestRoundRobinJedisPool.java

```diff
@@ -100,10 +98,35 @@ public FileVisitResult postVisitDirectory(Path dir, IOException exc) throws IOEx
         });
     }
 
-    private void addNode(String name, int port, String state) throws IOException,
-            InterruptedException, KeeperException {
-        ZooKeeper zk = new ZooKeeper("localhost:" + zkPort, 5000, null);
+    public ZooKeeper connect(String host, int timeout) throws Exception {
+        final CountDownLatch connectedLatch = new CountDownLatch(1);
+
+        try {
+            ZooKeeper zk = new ZooKeeper(host, timeout, new Watcher() {
+                @Override
+                public void process(WatchedEvent event) {
+                    if (event.getState() == Event.KeeperState.SyncConnected) {
+                        connectedLatch.countDown();
+                    }
+                }
+            });
+
+            if (ZooKeeper.States.CONNECTING == zk.getState()) {
+                connectedLatch.await();
+            }
+
+            return zk;
+        } catch (Exception e) {
+            return null;
+        }
+    }
+
+    private void addNode(String name, int port, String state) {
+        ZooKeeper zk = null;
+
         try {
+            zk = connect("localhost:" + zkPort, 5000);
```

Comment:
only tiny refactor old test code,  it is easier to create a new zk connection to do test.